proxy/proxy.go

```diff
@@ -298,10 +260,16 @@ func (p *proxy) setupProxyRequest(target *http.Request) {
 	}
 	target.URL.RawQuery = ""
 
-	handler.SetRequestXRequestStart(target)
+	setRequestXRequestStart(target)
 	target.Header.Del(router_http.CfAppInstance)
 }
 
+func setRequestXRequestStart(request *http.Request) {
+	if _, ok := request.Header[http.CanonicalHeaderKey("X-Request-Start")]; !ok {
+		request.Header.Set("X-Request-Start", strconv.FormatInt(time.Now().UnixNano()/1e6, 10))
```

Comment:
would [`time.Now().UnixMilli()`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/time/time.go;l=1222) not be equivalent and more readable in terms of what this timestamp actually represents?

Reply:
probably. but that's outside the scope of this PR. I've just moved the code needed as-is from the request handler to proxy.